src/pd/util.rs

```diff
@@ -293,6 +293,14 @@ where
             Ok(_) => true,
             // Error::Incompatible is returned by response header from pd, no need to retry
             Err(Error::Incompatible) => true,
+            Err(Error::Grpc(::grpc::Error::RpcFinished(err))) => {
+                // This means the old rpc connected with PD is finished (maybe PD restart),
+                // and this error will be retruned by the first heartbeat request after connecting
+                // which calls sender.send_all(). If retry, it may send the first heartbeat region
```

Comment:
Can you explain it in detail? I don't see any reason why it keep sending first heartbeat. Are you referring to last heartbeat?

Reply:
I do refer to the very first heartbeat after TiKV starting. You can check the description for this PR to see the reason.